src/camp_solver.c

```diff
@@ -629,13 +649,30 @@ int solver_run(void *solver_data, double *state, double *env, double t_initial,
 
   // Run the solver
   realtype t_rt = (realtype)t_initial;
+
   if (!sd->no_solve) {
+#ifdef CAMP_DEBUG_GPU
+  double starttimeCvode = MPI_Wtime();
+#endif
+#ifdef CAMP_USE_GPU
+    if(sd->use_cpu==1){
+      flag = CVode(sd->cvode_mem, (realtype)t_final, sd->y, &t_rt, CV_NORMAL);
+    }
+    else{
+      flag = cudaCVode(sd->cvode_mem, (realtype)t_final, sd->y,
+        &t_rt, sd);
+    }
+#else
     flag = CVode(sd->cvode_mem, (realtype)t_final, sd->y, &t_rt, CV_NORMAL);
+#endif
+#ifdef CAMP_DEBUG_GPU
+    sd->timeCVode += (MPI_Wtime() - starttimeCvode);
+#endif
     sd->solver_flag = flag;
-#ifndef FAILURE_DETAIL
+#ifdef FAILURE_DETAIL
```

Comment:
shouldn't this still be `#ifndef`?

Reply:
True. Anyway let me reverse the code to use a `ifdef` instead of `ifndef`.